certbot/plugins/dns_common.py

```diff
@@ -19,6 +20,24 @@
 logger = logging.getLogger(__name__)
 
 
+class OverrideChallengeAction(argparse.Action):
+    """Action class for parsing override-challenge."""
+
+    def __init__(self, dest, *args, **kwargs):
+        super(OverrideChallengeAction, self).__init__(*args, dest=dest, **kwargs)
+        self.challenge_map = {}
+        self.dest = dest
+        self.dest_map = dest + "_map"
+
+    def __call__(self, parser, namespace, override_chall, option_string=None):
+        for domain in namespace.domains:
+            self.challenge_map.setdefault(domain,
+                                          getattr(namespace, self.dest))
+        setattr(namespace, self.dest, override_chall)
```

Comment:
This probably should happen before the for loop above, as all the values in `self.challenge_map` are mapped to the default value of the parameter (`{acme}`) if not. This results to every domain getting the default value as an override domain.

Reply:
This part was actually as intended. The override option is intended to apply to domains that appear *after* it on the command line, so the loop here applies the *previous* value to *previous* domains. This is similar to what the webroot plugin does.

Reply:
(The new commit pushed today includes additional comments here to make the flow clearer).